### ISSUE TITLE
Package remains published if semantic-release fails

### ISSUE THREAD

**Issue by ThisIsMissEm** [semantic-release/npm#275]
Currently, if the semantic-release partially executes, the package remains published, even though it shouldn't.

This might be partially avoided if in `verifyConditions` we check if the package version we are about to publish has already been published, then we can fail early, aborting all future steps. However, other plugins failing to execute, hence failing the semantic-release also should revert the published package.

I'm seeking clarification on whether a version of a package can be unpublished within 72 hours from npm, as their policy is unclear: https://www.npmjs.com/policies/unpublish

https://twitter.com/thisismissem/status/1308723579085893632?s=21

**Comment by ThisIsMissEm** [comment]
In the future, this will be better fixed by the `--staged` flag: https://github.com/npm/rfcs/pull/92/files

**Comment by gr2m** [comment]
> This might be partially avoided if in `verifyConditions` we check if the package version we are about to publish has already been published

That's a great idea, I'd gladly accept a pull request which checks if the package version already exists in the registry! Would you like to get a pull request going yourself?

And yes, I'm following the `--staged` RFC with very big interest. Thank you Emilia

**Comment by travi** [comment]
> However, other plugins failing to execute, hence failing the semantic-release also should revert the published package.

I'm curious about the desire to revert the publish rather than recover from the failure and finish the rest of the publish process. While npm does allow a period where reverting is possible, I don't think we should encourage doing so since it can cause a number of problems for consumers. Personally, I'd rather see the registry be more immutable and encourage deprecation and publishing of follow up versions when legitimate bad publishes happen.

Could you describe the type of situation where you think it could be preferable to unpublish rather than leave the successfully published package but retry the failed steps that were supposed to follow?

**Comment by gr2m** [comment]
I'm not sure if you can unpublish a version for an established package, and then re-publish the same version, if I recall correctly that is not allowed

**Comment by travi** [comment]
> I recall correctly that is not allowed

good point. that is correct according to https://docs.npmjs.com/unpublishing-packages-from-the-registry

> **Note: package-name@version** is unique, and cannot be reused by unpublishing and re-publishing it. We recommend publishing a minor version update instead.
